### PR TITLE
Fixes issue #84 - Support multiple config files

### PR DESCRIPTION
**Pull request by asarkar**
Fixes issue #84 - Support multiple config files

### CODE REVIEW

File: main/java/io/github/azagniotov/stubby4j/yaml/YAMLMerger.java [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        Path configDir = config.toPath();
+        LOGGER.info("Config dir: {}.", configDir.toAbsolutePath());
+
+        if (!exists(configDir) || !isDirectory(configDir)) {
+            throw new IllegalArgumentException(
+                    String.format("%s doesn't exist or is not a directory.", configDir.toAbsolutePath())
+            );
+        }
+
+        List<Path> files = findAllYaml(configDir);
```
Comment (azagniotov) [inline main/java/io/github/azagniotov/stubby4j/yaml/YAMLMerger.java]: What is the need to actually write into a merged temp file? Why not to just merge loaded sub-files in memory? Also, did you test the merge functionality when `--watch` command line argument is provided - will the changes to one of the sub files cause a reload?

File: unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        File merged = YAMLMerger.merge(resources.toFile());
+        assertThat(merged).isNotNull();
+        assertThat(merged.exists() && merged.isFile() && merged.canRead()).isTrue();
+
+        for (String str : Arrays.asList("method: PUT", "status: 200")) {
+            long count = Files.lines(merged.toPath())
+                    .filter(line -> line.trim().equals(str))
+                    .count();
+            assertThat(count).isEqualTo(3L);
```
Comment (azagniotov) [inline unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java]: Can you please avoid from asserting for something in the loop?

File: unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        // IntelliJ and Gradle have different output directories
+        Path resources = root.resolveSibling("resources");
+        if (!exists(resources)) {
+            resources = root.getParent().resolveSibling("resources");
+        }
+        assertThat(exists(resources)).isTrue();
+
+        File merged = YAMLMerger.merge(resources.toFile());
+        assertThat(merged).isNotNull();
+        assertThat(merged.exists() && merged.isFile() && merged.canRead()).isTrue();
```
Comment (azagniotov) [inline unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java]: Can you please break the `boolean && boolean && boolean` up to three `assertThat` statements? If something fails, it is less clear what exactly failed

File: unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+import static com.google.common.truth.Truth.assertThat;
+import static java.nio.file.Files.exists;
+
+/**
+ * @author Abhijit Sarkar
+ */
+public class YAMLMergerTest {
+    @Test
+    public void testMerge() throws IOException, URISyntaxException {
```
Comment (azagniotov) [inline unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java]: This test is testing too much. Can you please break it up for several atomic tests that tests different parts of the behavior?

Comment (azagniotov) [inline unit/java/io/github/azagniotov/stubby4j/yaml/YAMLMergerTest.java]: It is OK for tests to `throws Exception` if you are not testing for specific exception

File: main/java/io/github/azagniotov/stubby4j/yaml/YAMLMerger.java [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+import static java.nio.file.StandardOpenOption.WRITE;
+import static java.util.stream.Collectors.toList;
+
+/**
+ * @author Abhijit Sarkar
+ */
+public final class YAMLMerger {
+    private static final Logger LOGGER = LoggerFactory.getLogger(YAMLMerger.class);
+    private static final PathMatcher YAML_MATCHER = FileSystems.getDefault()
+            .getPathMatcher("glob:**.{yaml,yml}");
```
Comment (azagniotov) [inline main/java/io/github/azagniotov/stubby4j/yaml/YAMLMerger.java]: There are no unit tests for `"glob:**.{yaml,yml}"`

**Comment by asarkar** [comment]
Sorry, I've exhausted the time I estimated for working on this CR. You can:
1.  Close it.
2. Take it as-is.
3. Use it as basis for making whatever changes you feel are required.

**Pull request closed without merging**